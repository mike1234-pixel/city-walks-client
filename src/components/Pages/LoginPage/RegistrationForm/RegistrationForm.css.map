{"version":3,"sourceRoot":"","sources":["RegistrationForm.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;EACA;;;AAGF;EACE;EACA;;;AAGF;EACE;;;AAGF;EACE;EACA;EACA","file":"RegistrationForm.css"}