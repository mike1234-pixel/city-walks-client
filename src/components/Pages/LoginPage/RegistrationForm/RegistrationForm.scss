/** @format */

.registration-header-container {
  width: 100%;
  text-align: center;
}

.registration-header {
  font-family: var(--logo-font);
  margin-top: 30px;
}

#request-activation-link {
  color: var(--blue);
}

.emphasis {
  background: rgba(63, 63, 63, 0.219);
  font-weight: bold;
  color: white;
}