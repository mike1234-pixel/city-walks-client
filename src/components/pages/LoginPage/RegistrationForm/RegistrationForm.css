/** @format */

.registration-header-container {
  width: 100%;
  text-align: center;
}

.registration-header {
  font-family: var(--display-font);
}

#request-activation-link {
  color: var(--blue);
}
