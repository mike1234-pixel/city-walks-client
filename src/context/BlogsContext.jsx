import { createContext, useState } from "react"

export const BlogsContext = createContext();

export const BlogsContextProvider = (props) => {

    const [blogPosts, setBlogPosts] = useState([])
    const [blogsLoading, setBlogsLoading] = useState(true)

    return (
        <BlogsContext.Provider 
            value={{
                blogPosts,
                setBlogPosts,
                blogsLoading,
                setBlogsLoading,
            }}
        >
            {props.children}
        </BlogsContext.Provider>
    )
}