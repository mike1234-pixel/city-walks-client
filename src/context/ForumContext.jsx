import { createContext, useState } from "react"
import BoardBox from "../components/pages/LoginPage/LoggedInView/Components/BoardBox/BoardBox"

export const ForumContext = createContext();

export const ForumContextProvider = (props) => {
 
    const [loadingBoards, setLoadingBoards] = useState(true)
    const [boards, setBoards] = useState([])
    const [selectedThreads, setSelectedThreads] = useState([])

    const [currentBoardName, setCurrentBoardName] = useState("")
    const [currentBoardId, setCurrentBoardId] = useState("")

    let displayBoards;

    if (!loadingBoards) {
    displayBoards = boards.map((board, index) => {
        return (
            <BoardBox boardId={board._id} name={board.name} description={board.description} index={index} key={index} />
                )
        })
        // create routes for boards with Thread as the view component
    }

    return (
        <ForumContext.Provider 
            value={{
                setBoards,
                boards, // data
                setLoadingBoards,
                loadingBoards,
                displayBoards,
                setSelectedThreads,
                selectedThreads, 
                setCurrentBoardName,
                currentBoardName,
                setCurrentBoardId,
                currentBoardId,
            }}>
            {props.children}
        </ForumContext.Provider>
    )
}

