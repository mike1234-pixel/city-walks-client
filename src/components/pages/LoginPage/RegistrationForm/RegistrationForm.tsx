import React from 'react'
import { MDBInput, MDBBtn, MDBIcon, MDBContainer } from "mdbreact"
import UserPortalNav from "../UserPortalNav"
import { Link } from "react-router-dom"
import axios, { AxiosError, AxiosResponse } from "axios"
import qs from "qs"
import { connect } from 'react-redux'
import { Dispatch } from 'redux'
import GlobalState from '../../../../types/State/Global/State'
import RegistrationCredentials from '../../../../types/PostRequests/RegistrationCredentials'
import './RegistrationForm.scss'

interface Props {
  firstName: string;
  lastName: string;
  registrationEmail: string;
  registrationPassword: string;
  activationMessageEmphasis: string;
  setFirstName: Function;
  setLastName: Function;
  setRegistrationEmail: Function;
  setRegistrationPassword: Function;
  setUserFirstName: Function;
  setUserLastName: Function;
  setActivationMessageEmphasis: Function;
}

const RegistrationForm: React.FC<Props> = (props: Props) => {

  const { firstName, lastName, registrationEmail, registrationPassword, activationMessageEmphasis, setFirstName, setLastName, setRegistrationEmail, setRegistrationPassword, setUserFirstName, setUserLastName, setActivationMessageEmphasis } = props

  const handleChangeRegistration: (event: React.ChangeEvent<any>) => void = (event) => {
    switch (event.target.name) {
      case "registration-fname":
        setFirstName(event.target.value)
        break;
      case "registration-lname":
        setLastName(event.target.value)
        break;
      case "registration-email":
        setRegistrationEmail(event.target.value)
        break;
      case "registration-password":
        setRegistrationPassword(event.target.value)
        break;
    }
  }

  const handleSubmitRegistration: (event: React.FormEvent) => void = (event) => {

    event.preventDefault()

    const payload: RegistrationCredentials = {
      fname: firstName,
      lname: lastName,
      email: registrationEmail,
      password: registrationPassword,
    };

    axios
      .post("https://city-walks.herokuapp.com/register-user", qs.stringify(payload))
      .then((res: AxiosResponse) => {
        if (res.data === "An account with this email already exists.") {
          alert("An account with this email already exists.")
        } else if (res.data === "We have sent you an email. Please verify your account by clicking the link in the mail.") {
          alert("We have sent you an email. Please verify your account by clicking the link in the email. (This code expires after 10 minutes)")
          setFirstName("")
          setLastName("")
          setRegistrationEmail("")
          setRegistrationPassword("")
          setUserFirstName(res.data.fname)
          setUserLastName(res.data.lname)
          setActivationMessageEmphasis("emphasis")
          window.scrollTo(0, 0)
        } else {
          alert("Registration Complete")
        }
      }).catch((err: AxiosError) => {
        console.log(err)
      });
  }


  return (
    <div key="user-registration">
      <UserPortalNav loggedIn={false} />
      <MDBContainer className="min-page-height">
        <div className="registration-header-container">
          <h2 className="registration-header">Register</h2>
          <p>Create a new account.</p>
          <div className={activationMessageEmphasis}>
            <p>You will receive an email containing an activation link to verify your email and activate your account.</p>
            <p>The activation link expires after 10 minutes.</p>
            <p>If the activation link expires, you can <Link id="request-activation-link" to="/forum/verify">request a fresh activation link.</Link></p>
          </div>
        </div>
        <form onSubmit={handleSubmitRegistration} className="user-registration-form display-form" key="user-registration-form">
          <MDBInput key="input-1" type="text" name="registration-fname" id="registration-fname" value={firstName} label="first Name" onChange={handleChangeRegistration} pattern="^[A-Za-z\-]+$" required />
          <MDBInput key="input-2" type="text" name="registration-lname" id="registration-lname" value={lastName} label="last Name" onChange={handleChangeRegistration} pattern="^[A-Za-z\-]+$" required />
          <MDBInput key="input-3" type="email" name="registration-email" id="registration-email" value={registrationEmail} label="email" onChange={handleChangeRegistration} required />
          <MDBInput key="input-4" type="password" name="registration-password" id="registration-password" value={registrationPassword} label="password (minimum 8 characters)" onChange={handleChangeRegistration} minLength="8" required />
          <MDBBtn outline color="elegant" type="submit">
            Register <MDBIcon icon="sign-in-alt" />
          </MDBBtn>
        </form>
      </MDBContainer>
    </div>
  )
}

const mapStateToProps: (state: GlobalState) => void = (state) => ({
  firstName: state.loginState.firstName,
  lastName: state.loginState.lastName,
  registrationEmail: state.loginState.registrationEmail,
  activationMessageEmphasis: state.loginState.activationMessageEmphasis
});

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    setFirstName: (firstName: string) => dispatch({ type: 'SET_FIRST_NAME', firstName }),
    setLastName: (lastName: string) => dispatch({ type: 'SET_NAME_NAME', lastName }),
    setRegistrationEmail: (registrationEmail: string) => dispatch({ type: 'SET_REGISTRATION_EMAIL', registrationEmail }),
    setRegistrationPassword: (registrationPassword: string) => dispatch({ type: 'SET_REGISTRATION_PASSWORD', registrationPassword }),
    setUserFirstName: (userFirstName: string) => dispatch({ type: 'SET_USER_FIRST_NAME', userFirstName }),
    setUserLastName: (userLastName: string) => dispatch({ type: 'SET_USER_LAST_NAME', userLastName }),
    setActivationMessageEmphasis: (activationMessageEmphasis: string) => dispatch({ type: 'SET_ACTIVATION_MESSAGE_EMPHASIS', activationMessageEmphasis }),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(RegistrationForm);