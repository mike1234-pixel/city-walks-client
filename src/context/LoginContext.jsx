import { createContext, useState } from "react"
import axios from "axios"
import qs from "qs"
import LoginForm from "../components/pages/LoginPage/LoginForm/LoginForm";
import RegistrationForm from "../components/pages/LoginPage/RegistrationForm/RegistrationForm"
import VerificationForm from "../components/pages/LoginPage/VerificationForm/VerificationForm";
import ResetPasswordForm from "../components/pages/LoginPage/ResetPasswordForm/ResetPasswordForm";
import ForgotPasswordForm from "../components/pages/LoginPage/ForgotPasswordForm/ForgotPasswordForm";
import Boards from "../components/pages/LoginPage/LoggedInView/Boards"

export const LoginContext = createContext();

export const LoginContextProvider = (props) => {
    const [popupVisible, setPopupVisible] = useState(false)

    const [firstName, setFirstName] = useState("")
    const [lastName, setLastName] = useState("")
    const [userId, setUserId] = useState("")
    const [registrationEmail, setRegistrationEmail] = useState("")
    const [registrationPassword, setRegistrationPassword] = useState("")
  
    const [loginEmail, setLoginEmail] = useState("")
    const [loginPassword, setLoginPassword] = useState("")
  
    // save all of this in local storage
    const [loggedIn, setLoggedIn] = useState(false)
    const [userFirstName, setUserFirstName] = useState("")
    const [userLastName, setUserLastName] = useState("")
  
    const [verificationEmail, setVerificationEmail] = useState("")

    const [resetPasswordEmail, setResetPasswordEmail] = useState("")
    const [resetPasswordOldPassword, setResetPasswordOldPassword] = useState("")
    const [resetPasswordNewPassword, setResetPasswordNewPassword] = useState("")
    const [resetPasswordConfirmNewPassword, setResetPasswordConfirmNewPassword] = useState("")

    const [forgotPasswordEmail, setForgotPasswordEmail] = useState("")

    const handlePopup = () => {
      setPopupVisible(false)
      localStorage.setItem("popupVisible", false)
    }

    // set display form on LoginPage
    const [form, setForm] = useState("loggedOutView");

    let displayForm = <Boards/>;
    if (form === "boards") {
      displayForm = <Boards/>;
    } else if (form === "loginForm") {
      displayForm = <LoginForm />;
    } else if (form === "registrationForm") {
      displayForm = <RegistrationForm />;
    } else if (form === "verificationForm") {
      displayForm = <VerificationForm />;
    } else if (form === "resetPasswordForm") {
      displayForm = <ResetPasswordForm />;
    } else if (form === "forgotPasswordForm") {
      displayForm = <ForgotPasswordForm />;
    }

    const handleChangeRegistration = (event) => {
        switch(event.target.name) {
            case "registration-fname":
              setFirstName(event.target.value)
              break;
            case "registration-lname":
              setLastName(event.target.value)
              break;
            case "registration-email":
              setRegistrationEmail(event.target.value)
              break;
            case "registration-password":
              setRegistrationPassword(event.target.value)
              break;
          } 
    }
  
    const handleSubmitRegistration = (event) => {
      console.log("handle submit triggered")
        event.preventDefault()
  
        const payload = {
            fname: firstName,
            lname: lastName,
            email: registrationEmail,
            password: registrationPassword,
          };
  
    axios
        .post("http://localhost:5000/register-user", qs.stringify(payload))
        .then((res, err) => {
          if (err) {
            console.log(err);
          } else if (res.data === "An account with this email already exists.") {
            alert("An account with this email already exists.")
          } else if (res.data === "We have sent you an email. Please verify your account by clicking the link in the mail.") {
            alert("We have sent you an email. Please verify your account by clicking the link in the email. (This code expires after 10 minutes)")
           } else {
            alert("Registration Complete")
            setLoggedIn(true)
            setFirstName("")
            setLastName("")
            setRegistrationEmail("")
            setRegistrationPassword("")
            setUserFirstName(res.data.fname)
            setUserLastName(res.data.lname)
            window.scrollTo(0, 0)
          }
        });
    }
  
    const handleChangeLogin = (event) => {
      switch(event.target.name) {
          case "login-email":
            setLoginEmail(event.target.value)
            break;
          case "login-password":
            setLoginPassword(event.target.value)
            break;
        } 
  }
  
  const handleSubmitLogin = (event) => {
    console.log("handle submit login triggered")
      event.preventDefault()
  
      const payload = {
          email: loginEmail,
          password: loginPassword
        };
  
  axios
      .post("http://localhost:5000/login-user", qs.stringify(payload))
      .then((res, err) => {
        if (err) {
          console.log(err);
        } else if (res.data === "Your account exists but is not activated. Please click 'verify account' for email verification.") {
          alert("Your account exists but is not activated. Please click 'verify account' for email verification.")
        } else if (res.data === "unsuccessful login attempt") {
          alert("Unsuccessful Login Attempt. Please Try Again.")
        } else {
          alert("Login Complete")
          setLoggedIn(true)
          setLoginEmail("")
          setLoginPassword("")
          setForm("")
          setUserId(res.data._id)
          setUserFirstName(res.data.fname)
          setUserLastName(res.data.lname)
          localStorage.setItem("loggedIn", true)
          localStorage.setItem("userId", res.data._id)
          localStorage.setItem("userFirstName", res.data.fname)
          window.scrollTo(0, 0)
        }
      });
  }

  // resend verification email

  const handleChangeVerification = (event) => {
    switch(event.target.name) {
        case "verification-email":
          setVerificationEmail(event.target.value)
          break;
      }   
  }

const handleSubmitVerification = (event) => {
    console.log("handle submit verification")
    event.preventDefault()

    const payload = {
        email: verificationEmail,
      };

axios
    .post("http://localhost:5000/reverify-user", qs.stringify(payload))
    .then((res, err) => {
      if (err) {
        console.log(err);
      } else {
        alert("Verification email sent. Check your inbox.")
        setVerificationEmail("")
        window.scrollTo(0, 0)
      }
    })
}

  // reset password

  const handleChangeResetPassword = (event) => {
    switch(event.target.name) {
        case "reset-email":
          setResetPasswordEmail(event.target.value)
          break;
        case "old-password":
          setResetPasswordOldPassword(event.target.value)
          break;
        case "new-password":
          setResetPasswordNewPassword(event.target.value)
          break;
        case "confirm-new-password":
          setResetPasswordConfirmNewPassword(event.target.value)
          break;
      }   
  }

const handleSubmitResetPassword = (event) => {
    console.log("handle submit reset password")
    event.preventDefault()

    const payload = {
        email: resetPasswordEmail,
        oldPassword: resetPasswordOldPassword,
        newPassword: resetPasswordNewPassword,
      };

      if (resetPasswordNewPassword === resetPasswordConfirmNewPassword) {
        axios
        .post("http://localhost:5000/reset-password-with-old-password", qs.stringify(payload))
        .then((res, err) => {
          if (err) {
            console.log(err);
          } else if (res.data === "user not found") {
            alert("We could not find your account. Please try again.")
          } else if (res.data === "old password does not match password in the database") {
            alert("The password you entered does not match the account. Please try again or use the Forgot Password option.")
          } else if (res.data === "password successfully updated") {
            alert("Password reset. You can now login.")
            setResetPasswordEmail("")
            setResetPasswordOldPassword("")
            setResetPasswordNewPassword("")
            setResetPasswordConfirmNewPassword("")
            window.scrollTo(0, 0)
          }
        })
      } else {
        alert("Passwords don't match. Please try again.")
      }

}

  // forgot password

  const handleChangeForgotPassword = (event) => {
    switch(event.target.name) {
        case "forgot-password-email":
          setForgotPasswordEmail(event.target.value)
          break;
      }   
  }

const handleSubmitForgotPassword = (event) => {
    console.log("handle submit forgot password")
    event.preventDefault()

    const payload = {
        email: forgotPasswordEmail,
      };

axios
    .post("http://localhost:5000/forgot-password", qs.stringify(payload))
    .then((res, err) => {
      if (err) {
        console.log(err);
      } else {
        alert("We have sent you an email. Please click the click in your email to reset your password")
        setForgotPasswordEmail("")
        window.scrollTo(0, 0)
      }
    })
}
  
  const logOut = () => {
    localStorage.clear()
    localStorage.setItem("popupVisible", false)
    setLoggedIn(false)
    setUserId("")
    setUserFirstName("")
    setUserLastName("")
    setForm("boards")
  }

    return (
        <LoginContext.Provider 
            value={{
              popupVisible,
              setPopupVisible,
              handlePopup,
               // display form
               displayForm,
               setForm,
               // logged in state 
               loggedIn,
               // registration state and functions
                firstName,
                lastName,
                registrationEmail,
                registrationPassword,
                handleChangeRegistration,
                handleSubmitRegistration,
                // login state and functions
                loginEmail,
                loginPassword,
                handleChangeLogin: handleChangeLogin,
                handleSubmitLogin: handleSubmitLogin,
                setLoggedIn,
                setUserFirstName,
                setUserId,
                // logged in user data
                userFirstName,
                userId,
                userLastName,
                // resend verification email
                verificationEmail,
                handleChangeVerification,
                handleSubmitVerification,
                // reset password
                resetPasswordEmail,
                resetPasswordOldPassword,
                resetPasswordNewPassword,
                resetPasswordConfirmNewPassword,
                handleChangeResetPassword,
                handleSubmitResetPassword,
                // forgot password
                forgotPasswordEmail,
                handleChangeForgotPassword,
                handleSubmitForgotPassword,
                // logout function
                logOut
            }}
        >
            {props.children}
        </LoginContext.Provider>
    )
}

// // setter
// localStorage.setItem('myData', data);
 
// // getter
// localStorage.getItem('myData');
 
// // remove
// localStorage.removeItem('myData');
 
// // remove all
// localStorage.clear();

