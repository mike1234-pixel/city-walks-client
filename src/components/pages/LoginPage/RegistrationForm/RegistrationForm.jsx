import { useContext } from 'react'
import { LoginContext } from '../../../../context/LoginContext'
import { MDBInput, MDBBtn, MDBIcon } from "mdbreact"
import './RegistrationForm.css'

const RegistrationForm = () => {

  const {
        firstName, 
        lastName, 
        registrationEmail, 
        registrationPassword, 
        handleChangeRegistration,
        handleSubmitRegistration,
        setForm
    } = useContext(LoginContext)


    return (
    <div key="user-registration">
      <div className="registration-header-container">
        <h2 className="registration-header">Register</h2>
        <p>Create a new account.</p>
        <p>You will receive an email containing an activation link to verify your email and activate your account.</p>
        <p>The activation link expires after 10 minutes.</p>
        <p>If the activation link expires, you can <a id="request-activation-link" onClick={() => setForm("verificationForm")}>request a fresh activation link.</a></p>
      </div>
        <form onSubmit={handleSubmitRegistration} className="user-registration-form display-form" key="user-registration-form">
            <MDBInput key="input-1" type="text" name="registration-fname" id="registration-fname" value={firstName} label="first Name" onChange={handleChangeRegistration} pattern="^[A-Za-z\-]+$" required/>
            <MDBInput key="input-2" type="text" name="registration-lname" id="registration-lname" value={lastName} label="last Name" onChange={handleChangeRegistration} pattern="^[A-Za-z\-]+$" required/>
            <MDBInput key="input-3" type="email" name="registration-email" id="registration-email" value={registrationEmail} label="email" onChange={handleChangeRegistration} required/>
            <MDBInput key="input-4" type="password" name="registration-password" id="registration-password" value={registrationPassword} label="password (minimum 8 characters)" onChange={handleChangeRegistration} minLength="8" required/>
            <MDBBtn outline color="elegant" type="submit">
                Register <MDBIcon icon="sign-in-alt" />
          </MDBBtn>
        </form>
    </div>
    )
}

export default RegistrationForm