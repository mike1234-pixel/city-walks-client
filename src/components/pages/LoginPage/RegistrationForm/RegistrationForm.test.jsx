import renderer from 'react-test-renderer'
import { LoginContextProvider } from "../../../../context/LoginContext"
import RegistrationForm from "./RegistrationForm"

// snapshot test
it('RegistrationForm renders correctly', () => {  

  const tree = renderer
    .create(
            <LoginContextProvider>
                <RegistrationForm />
            </LoginContextProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});