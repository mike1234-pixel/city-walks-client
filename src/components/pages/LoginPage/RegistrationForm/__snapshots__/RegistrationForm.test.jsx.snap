// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm renders correctly 1`] = `
<div>
  <div
    className="registration-header-container"
  >
    <h2
      className="registration-header"
    >
      Register
    </h2>
    <p>
      Create a new account.
    </p>
    <p>
      You will receive an email containing an activation link to verify your email and activate your account.
    </p>
    <p>
      The activation link expires after 10 minutes.
    </p>
    <p>
      If the activation link expires, you can 
      <a
        id="request-activation-link"
        onClick={[Function]}
      >
        request a fresh activation link.
      </a>
    </p>
  </div>
  <form
    className="user-registration-form display-form"
    onSubmit={[Function]}
  >
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="registration-fname"
        name="registration-fname"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        pattern="^[A-Za-z\\\\-]+$"
        required={true}
        type="text"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="registration-fname"
        id=""
        onClick={[Function]}
      >
        first Name
      </label>
    </div>
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="registration-lname"
        name="registration-lname"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        pattern="^[A-Za-z\\\\-]+$"
        required={true}
        type="text"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="registration-lname"
        id=""
        onClick={[Function]}
      >
        last Name
      </label>
    </div>
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="registration-email"
        name="registration-email"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="email"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="registration-email"
        id=""
        onClick={[Function]}
      >
        email
      </label>
    </div>
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="registration-password"
        minLength="8"
        name="registration-password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="password"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="registration-password"
        id=""
        onClick={[Function]}
      >
        password (minimum 8 characters)
      </label>
    </div>
    <button
      className="btn-elegant btn-outline-elegant btn Ripple-parent"
      data-test="button"
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="submit"
    >
      Register 
      <i
        className="fa fa-sign-in-alt"
        data-test="fa"
      />
      <div
        className="Ripple Ripple-outline "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </button>
  </form>
</div>
`;
